Doc/reference/lexical_analysis.rst

```diff
@@ -53,6 +53,9 @@ form using the ASCII sequence CR LF (return followed by linefeed), or the old
 Macintosh form using the ASCII CR (return) character.  All of these forms can be
 used equally, regardless of platform.
 
+The end of a file may also serve as a terminator for a physical line. This is
+relevant when a file does not end with an end-of-line sequence.
```

Comment:
Python source can be a string as will as a file, and strings are more likely to be missing \n.  So I prefer 'Python input' or 'Python source' to 'a file'.   I think 'source files' above should be 'source strings or files'.

 I actually think the addition should be a single sentence added to the preceding paragraph. 

"The end of input also acts as an implied end-of-line.